### PR TITLE
add tests for compat_v1 encdoing

### PR DESCRIPTION
**Pull request by niedhui**
###  What have you changed?

Add some tests for compat_v1  encoding.

###  What is the type of the changes?

 Engineering 

###  How is the PR tested?

- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?
NO 

###  Refer to a related PR or issue link (optional)
#6132

### CODE REVIEW

**Comment by AndreMouche** [comment]
Please fix conflicts, thank you

**Comment by AndreMouche** [comment]
@HunDunDM

File: components/tidb_query/src/codec/row/v2/encoder.rs [hunk 1]
```diff
@@ -49,6 +49,7 @@ pub struct Column {
     ft: FieldType,
 }
 
+// TODO: refactor
```
Comment (MyonKeminta) [inline components/tidb_query/src/codec/row/v2/encoder.rs]: Please explain it a little more clear about how should we refactor it.

Comment (niedhui) [reply components/tidb_query/src/codec/row/v2/encoder.rs]: sorry, forgot to delete this line 😄

**Pull request merged**